src/test/scala/org/neo4j/spark/DataSourceReaderNeo4j35xTSE.scala

```diff
@@ -0,0 +1,37 @@
+package org.neo4j.spark
+
+import org.apache.commons.lang3.exception.ExceptionUtils
+import org.apache.spark.SparkException
+import org.junit.Assert.{assertEquals, assertTrue, fail}
+import org.junit.{Assume, BeforeClass, Ignore, Test}
+import org.neo4j.driver.exceptions.ClientException
+
+object DataSourceReaderNeo4j35xTSE {
+  @BeforeClass
+  def checkNeo4jVersion() {
+    Assume.assumeTrue(TestUtil.neo4jVersion().startsWith("3.5"))
+  }
+}
+
+class DataSourceReaderNeo4j35xTSE extends SparkConnectorScalaBaseTSE {
```

Comment:
Another unit test scenario here which is the same problem but a different source, please add this and see if it behaves:

User connects to Neo4j 4.0 with databases (neo4j, system) and issues a query with `.option("database", "noexist")`.   Actually it's more likely they'll have like a typo in the database or whatever.

In 3.5 there is no valid database option, but in 4.0 you can always mess it up.  ;)